onnx/mapping.py

```diff
@@ -1,53 +1,51 @@
 # SPDX-License-Identifier: Apache-2.0
 
-from onnx import TensorProto, SequenceProto, OptionalProto
+import warnings
+from typing import Any
 import numpy as np  # type: ignore
+from onnx import TensorProto, SequenceProto, OptionalProto
 
-# This map is used for converting TensorProto values into Numpy arrays
-TENSOR_TYPE_TO_NP_TYPE = {
-    int(TensorProto.FLOAT): np.dtype('float32'),
-    int(TensorProto.UINT8): np.dtype('uint8'),
-    int(TensorProto.INT8): np.dtype('int8'),
-    int(TensorProto.UINT16): np.dtype('uint16'),
-    int(TensorProto.INT16): np.dtype('int16'),
-    int(TensorProto.INT32): np.dtype('int32'),
-    int(TensorProto.INT64): np.dtype('int64'),
-    int(TensorProto.BOOL): np.dtype('bool'),
-    int(TensorProto.FLOAT16): np.dtype('float16'),
-    int(TensorProto.BFLOAT16): np.dtype('float32'),  # Native numpy does not support bfloat16 so now use float32 for bf16 values
-    int(TensorProto.DOUBLE): np.dtype('float64'),
-    int(TensorProto.COMPLEX64): np.dtype('complex64'),
-    int(TensorProto.COMPLEX128): np.dtype('complex128'),
-    int(TensorProto.UINT32): np.dtype('uint32'),
-    int(TensorProto.UINT64): np.dtype('uint64'),
-    int(TensorProto.STRING): np.dtype('object')
+# tensor_dtype: (numpy type, storage type, string name)
+TENSOR_TYPE_MAP = {
+    int(TensorProto.FLOAT): (np.dtype('float32'), int(TensorProto.FLOAT), 'TensorProto.FLOAT'),
```

Comment:
maybe a namedtuple would help to remember the meaning of each element

Reply:
Great idea. I have introduced `TensorDtypeMap = NamedTuple(
    "TensorDtypeMap", [("np_type", np.dtype), ("storage_type", int), ("name", str)]
)` to better manage this data structure.